src/pykoala/corrections/sky.py

```diff
@@ -71,443 +187,640 @@ def pol_continuum(data, wavelength, pol_order=3, **polfit_kwargs):
 # =============================================================================
-# TODO: convert this class to an ABC
 
 class SkyModel(object):
     """
-    Abstract class of a sky emission model.
+    Abstract class for a sky emission model.
 
     Attributes
     ----------
-    wavelength:
-        TODO
-    intensity:
-        TODO 1D or 2D array, default None
-    variance:
-        TODO
-    verbose:
-        Print messages during the execution.
+    wavelength : np.ndarray
+        1-D array representing the wavelengths of the sky model.
+    intensity : np.ndarray
+        Array representing the intensity of the sky model.
+    variance : np.ndarray
+        Array representing the variance associated with the sky model. 
+        It must have the same dimensions as `intensity`.
+    verbose : bool, optional
+        If True, print messages during execution. Default is True.
+
     Methods
     -------
-    substract
-    substract_PCA
+    subtract(data, variance, axis=-1)
+        Subtracts the sky model from the given data.
+    
+    subtract_pca()
+        Placeholder for PCA subtraction method.
+    
+    vprint(*messages)
+        Print messages if `verbose` is True.
     """
+
     verbose = True
 
     def __init__(self, **kwargs):
+        """
+        Initialize the SkyModel object.
+
+        Parameters
+        ----------
+        **kwargs : dict, optional
+            Dictionary of parameters to initialize the SkyModel.
+            Accepted keys are:
+            - wavelength : np.ndarray
+                1-D array representing the wavelengths of the sky model.
+            - intensity : np.ndarray
+                Array representing the intensity of the sky model.
+            - variance : np.ndarray
+                Array representing the variance associated with the sky model.
+            - verbose : bool
+                If True, print messages during execution. Default is True.
+        """
         self.wavelength = kwargs.get('wavelength', None)
         self.intensity = kwargs.get('intensity', None)
         self.variance = kwargs.get('variance', None)
         self.verbose = kwargs.get('verbose', True)
 
     def substract(self, data, variance, axis=-1, verbose=False):
-        """Substracts the sky_model to all fibres in the rss
+        """
+        Subtracts the sky model from the given data.
 
         Parameters
         ----------
-        data: (np.ndarray)
-            Data array for which the sky will be substracted
-        variance: (np.ndarray)
-            Array of variance data to include errors on determining the sky.
-        axis: (np.ndarray)
-            Spectral direction of data
+        data : np.ndarray
+            Data array from which the sky will be subtracted.
+        variance : np.ndarray
+            Array of variance data to include errors in determining the sky.
+        axis : int, optional
+            Spectral direction of the data. Default is -1.
+
         Returns
```

Comment:
Notwithstanding my comment above, the actual implementation of this method looks somewhat weird to me. Shouldn't the argument be a `DataContainer`?

Reply:
Actually, I am not entirely sure what you meant in your previous comment XD. I thought about this method as a lower lever step that merely involves subtracting the background from an array. I reckon you are right, this is rather redundant as we would only applied it to DataContainers. Maybe it is worth doing some refactoring...

Reply:
This method implements sky subtraction... from a generic pair of `intensity` and `variance` arrays!
We are clearly talking about an `RSS`, `Cube`, or `SpectraContainer` here (I don't really like the `DataContainer` if this class may eventually include other kinds of data in the future).

What I meant in my comment above is that `SkyModel.subtract(rss)` is pretty much the same as `SkySubsCorrection.apply(rss)`, and therefore, according to the Highlander principle, there can only be one.
Which one is another question; I'd actually decapitate all the *Correction classes, but I'm fine with sticking with them.

For now ;^D